cms/djangoapps/contentstore/views/item.py

```diff
@@ -1453,6 +1456,25 @@ def _get_release_date(xblock, user=None):
     return get_default_time_display(xblock.start) if xblock.start != DEFAULT_START_DATE else None
 
 
+def validate_due_date(xblock, user=None):
+    """
+    Validates the due date for the xblock, resetting if invalid due date provided
+    """
+
+    set_to_none = False
+    try:
+        set_to_none = xblock.due.year < 1900
+
+    # For cases when due date is None
+    except AttributeError:
+        pass
+
+    if set_to_none and user:
```

Comment:
Why `user` is required?

Reply:
Thanks for mentioning this. There is no need for user now as the function call location has been changed and this function is now only changing due date, but not updating it in the modulestore.

Reply:
and I have simplified this function. will update the PR to reflect the change.

Reply:
@awaisdar001